src/replica/replica_stub.cpp

```diff
@@ -389,9 +451,40 @@ void replica_stub::initialize(const replication_options &opts, bool clear /* = f
         }
     }
 
+    std::string server_key;
+    dsn::replication::kms_info kms_info;
+    std::string kms_path = utils::filesystem::path_combine(
+        _options.data_dirs[0], dsn::replication::kms_info::kKmsInfo);
+    if (FLAGS_encrypt_data_at_rest && !utils::is_empty(FLAGS_hadoop_kms_url)) {
+        key_provider.reset(new dsn::security::KMSKeyProvider(
+            ::absl::StrSplit(FLAGS_hadoop_kms_url, ",", ::absl::SkipEmpty()), FLAGS_cluster_name));
+        auto ec = dsn::utils::load_rjobj_from_file(
+            kms_path, dsn::utils::FileDataType::kNonSensitive, &kms_info);
+        if (ec != dsn::ERR_OK) {
```

Comment:
Use the dsn::ERR_PATH_NOT_FOUND code to check the first time starting the server.
LOG fatal and exit the process in other error code.